highfidelity/hifi
# Fix overlay rendering to use cached Render Transform information

This pr touches all the Overlay 3Ds to make sure they capture their _renderTransform (or extra properties representing their position) during the GameLoop update. and use theese captured render tranform during rendering calls.

This PR goes into game-render-interface branch where we consolidate these cahgnges accross the code of Interface.

## CODE REVIEW HINTS
I have changed in every  3DOverlay::render calls the way the transform or transform coordinates are passed to the batch to be coming from the cached variables _renderTransform.
the _renderxxx elements are all computed in Game loop from the evalRenderTransform call

which is itself called in the update of the overlays if needed and then passed as value to the render items.
To actually trigger this to happen, we call "notifyRenderTransformChange" anywhere this needs to be recomputed (when dimension changes, or position or...)

At the end of the day, all the variables needed to draw the overlay elements are computed in Game loop, not in Render loop

## TEST PLAN
- make sure that the overlay elements are rendering correctly at the right position and without latency
- check that the tablet web overlay is moving together correctly with the tablet Model
- using edit mode, check that the overlay / wdgets are behaving correclty.
 - make sure they are all visible as expected
 - make sure they behave correctly when modified or when the inspected object is moved

## KNOWN ISSUE
- When using the tablet in desktop, the tablet is showing position tearing when moving around with the avatar
